src/bindgen/ir/function.rs

```diff
@@ -185,6 +191,33 @@ impl Function {
                 reserved::escape(name);
             }
         }
+
+        // Rename pointer arguments which need to use the C-array notation
+        if let Some(tuples) = self.annotations.list("ptrs-as-arrays") {
+            let mut ptrs_as_arrays: HashMap<String, String> = HashMap::new();
+            for str_tuple in tuples {
+                let parts: Vec<&str> = str_tuple[1..str_tuple.len() - 1]
+                    .split(';')
+                    .map(|x| x.trim())
+                    .collect();
+                debug_assert!(parts.len() == 2);
+                ptrs_as_arrays.insert(parts[0].to_string(), parts[1].to_string());
+            }
+
+            for args in &mut self.args {
+                if let Some(value) = ptrs_as_arrays.get(args.name.as_ref().unwrap()) {
+                    let (ptr_as_array, is_const) = match &args.ty {
```

Comment:
This seems like it may not be the best place to do this conversion... the constness shenanigans look wrong, constness is only about the C declarator and it's needed because of the automatic decay to pointer... But perhaps it's fine.

Still the `is_const_annotation` field should be properly documented. The name reads like the type has some sort of extra `const` annotation somewhere, but this is just about whether the pointer this array represents should be qualified as const.

Reply:
> This seems like it may not be the best place to do this conversion... the constness shenanigans look wrong, constness is only about the C declarator and it's needed because of the automatic decay to pointer... But perhaps it's fine.

I think it's the best place to do this conversion, otherwise we need to move around the length of each array and that would be confusing in my opinion

> Still the `is_const_annotation` field should be properly documented. The name reads like the type has some sort of extra `const` annotation somewhere, but this is just about whether the pointer this array represents should be qualified as const.

I'm going to change that name, effectively it is not so good as it seems.

Reply:
Nevermind, perhaps I found another way